plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/AbstractFBErrorMarkerResolution.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.marker.resolution;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.fordiac.ide.model.libraryElement.FB;
+import org.eclipse.fordiac.ide.model.libraryElement.FBType;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibrary;
+import org.eclipse.fordiac.ide.model.typelibrary.impl.ErrorFBTypeEntryImpl;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.ui.views.markers.WorkbenchMarkerResolution;
+
+public abstract class AbstractFBErrorMarkerResolution extends WorkbenchMarkerResolution {
+
+	private final String label;
+	private final IMarker marker;
+
+	protected AbstractFBErrorMarkerResolution(final String label, final IMarker marker) {
+		this.label = label;
+		this.marker = marker;
+	}
+
+	public TypeLibrary getTypeLibrary() {
+		try {
+			final EObject target = FordiacErrorMarker.getTarget(marker);
+			if (EcoreUtil.getRootContainer(target) instanceof final LibraryElement le) {
+				return le.getTypeLibrary();
+			}
+		} catch (final IllegalArgumentException | CoreException e) {
+			// target not found
+		}
+		return null;
+	}
+
+	@Override
+	public String getLabel() {
+		return label;
+	}
+
+	@Override
+	public String getDescription() {
+		return getLabel();
+	}
+
+	@Override
+	public Image getImage() {
+		return null;
+	}
+
+	@Override
+	public IMarker[] findOtherMarkers(final IMarker[] markers) {
+		try {
+			final EObject target = FordiacErrorMarker.getTarget(marker);
+			if (target instanceof final ErrorMarkerFBNElement ref) {
+				return findOtherTypeMarker(markers, ref.getType());
+			}
+			if (target instanceof final FB fb && fb.getTypeEntry() instanceof final ErrorFBTypeEntryImpl ref) {
+				return findOtherTypeMarker(markers, ref.getType());
+			}
+		} catch (final IllegalArgumentException | CoreException e) {
+			// target not found
+		}
+		return new IMarker[0];
```

Comment:
I have created a PR https://github.com/eclipse-4diac/4diac-ide/pull/421 that adds additional information to error markers, which should make it easier to find other markers, particularly without relying on `FordiacErrorMarker.getTarget(...)`:
```suggestion
				return Stream.of(markers)
				.filter(other -> LibraryElementValidator.DIAGNOSTIC_SOURCE.equals(FordiacErrorMarker.getSource(other))
						&& LibraryElementValidator.TYPED_CONFIGUREABLE_OBJECT__VALIDATE_TYPE == FordiacErrorMarker
								.getCode(other)
						&& Arrays.equals(FordiacErrorMarker.getData(other), FordiacErrorMarker.getData(marker)))
				.toArray(IMarker[]::new);
```